core/src/main/java/org/elasticsearch/index/query/HasChildQueryBuilder.java

```diff
@@ -76,40 +72,23 @@
     private static final ParseField INNER_HITS_FIELD = new ParseField("inner_hits");
 
     private final QueryBuilder<?> query;
-
     private final String type;
+    private final ScoreMode scoreMode;
+    private final InnerHitBuilder innerHitBuilder;
+    private int minChildren;
+    private int maxChildren;
```

Comment:
* I think score_mode should be a constructor parameter, as it is important to always think about that.
* I'll move inner hits to a setter.
* I'll bring back the defaults for min and max kids.